src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -135,6 +133,29 @@ private String[] openDataListEntrySelector(Function<Workspace, List<DataListEntr
 		return (String[]) Platform.enterNestedEventLoop(NESTED_LOOP_KEY);
 	}
 
+	/**
+	 * Opens a string selector window, where each string has a value that is returned by selecting it
+	 *
+	 * @param map  {"readable name": value}
+	 * @param type The type of the data list, used for the selector title and message
+	 * @return A {"value", "readable name"} pair, or the default entry if no entry was selected
+	 */
+
+	private String[] openKeyAndValueEntrySelector(Map<String, String> map, String type) {
+		SwingUtilities.invokeLater(() -> {
+			String[] retval = new String[] { "", L10N.t("blockly.extension.data_list_selector.no_entry") };
+			Pair<String, String> selected = KeyAndValueSelectorDialog.openSelectorDialog(mcreator, map,

```

Comment:
We have Tuple in the codebase:

![image](https://github.com/user-attachments/assets/52426ab2-044d-4df7-abb8-7efaacee81af)